geowave/src/main/scala/geotrellis/spark/io/geowave/GeowaveAttributeStore.scala

```diff
@@ -75,14 +74,15 @@ object GeowaveAttributeStore {
   }
 
   def adapters(bao: BasicAccumuloOperations): Array[RasterDataAdapter] = {
-    val as = new AccumuloAdapterStore(bao).getAdapters
-    val list = Iterator.iterate(as)({ _ => as })
-      .takeWhile({ _ => as.hasNext })
-      .map(_.next.asInstanceOf[RasterDataAdapter])
-      .toArray
-
-    as.close
-    list
+    for (adapters <- managed(new AccumuloAdapterStore(bao).getAdapters)) {
+      val list = Iterator.iterate(adapters)({ _ => adapters })
+        .takeWhile({ _ => adapters.hasNext })
+        .map(_.next.asInstanceOf[RasterDataAdapter])
```

Comment:
Here, I made the (temporary) assumption that all of the adapters are `RasterDataAdapters`.  Is that a safe assumption given the fact that these adapters all come from one GeoWave namspace, or should they be filtered?  If the latter, what is the preferred way to do that?